### PR TITLE
RCNN model API finalization.

### PR DESCRIPTION
**Pull request by tanzhenyu**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [ ] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: keras_cv/models/object_detection/faster_rcnn.py [hunk 1]
```diff
@@ -334,7 +308,7 @@ def __init__(
         )
         self.roi_pooler = _ROIAligner(bounding_box_format="yxyx")
         self.rcnn_head = rcnn_head or RCNNHead(classes)
-        self.backbone = backbone or _resnet50_backbone(include_rescaling)
+        self.backbone = backbone
```
Comment (ianstenbit) [inline keras_cv/models/object_detection/faster_rcnn.py:311]: To me, this means that `backbone` is required, and we shouldn't have a default value -- right?

**Review by ianstenbit — CHANGES_REQUESTED** [review]

File: keras_cv/models/object_detection/faster_rcnn.py (hunk 1 above)
Comment (tanzhenyu) [reply keras_cv/models/object_detection/faster_rcnn.py:311]: Maybe let's do `backbone or keras_cv.models.ResNet50(include_rescaling=True)`?

Comment (ianstenbit) [reply keras_cv/models/object_detection/faster_rcnn.py:311]: that's fine with me

**Review by ianstenbit — APPROVED** [review]
LGTM after default backbone change

File: keras_cv/models/object_detection/faster_rcnn.py (hunk 1 above)
Comment (ianstenbit) [reply keras_cv/models/object_detection/faster_rcnn.py:311]: (Will also need include_top=False and .as_backbone)

**Comment by tanzhenyu** [comment]
Force merging to test RetinaNet in https://github.com/keras-team/keras-cv/pull/1218

**Pull request merged**